### PR TITLE
HBASE-28389 HBase backup yarn queue parameter ignored

### PR DESCRIPTION
**Pull request by 2005hithlj**
https://issues.apache.org/jira/browse/HBASE-28389

### CODE REVIEW

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 15s |  |  Docker mode activated.  |
|||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  |  No case conflicting files found.  |
| +0 :ok: |  codespell  |   0m  0s |  |  codespell was not available.  |
| +0 :ok: |  detsecrets  |   0m  0s |  |  detect-secrets was not available.  |
| +1 :green_heart: |  @author  |   0m  0s |  |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  |  Patch does not have any anti-patterns.  |
|||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   3m  4s |  |  master passed  |
| +1 :green_heart: |  compile  |   0m 32s |  |  master passed  |
| +1 :green_heart: |  checkstyle  |   0m 12s |  |  master passed  |
| +1 :green_heart: |  spotbugs  |   0m 33s |  |  master passed  |
| +1 :green_heart: |  spotless  |   0m 45s |  |  branch has no errors when running spotless:check.  |
|||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   2m 54s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 30s |  |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 30s |  |  the patch passed  |
| +1 :green_heart: |  blanks  |   0m  0s |  |  The patch has no blanks issues.  |
| +1 :green_heart: |  checkstyle  |   0m 11s |  |  the patch passed  |
| +1 :green_heart: |  spotbugs  |   0m 37s |  |  the patch passed  |
| +1 :green_heart: |  hadoopcheck  |  10m 31s |  |  Patch does not cause any errors with Hadoop 3.3.6 3.4.0.  |
| +1 :green_heart: |  spotless  |   0m 44s |  |  patch has no errors when running spotless:check.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 11s |  |  The patch does not generate ASF License warnings.  |
|  |   |  27m 25s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.46 ServerAPI=1.46 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6131/1/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/6131 |
| Optional Tests | dupname asflicense javac spotbugs checkstyle codespell detsecrets compile hadoopcheck hbaseanti spotless |
| uname | Linux 84b5b1f1b9b9 5.4.0-182-generic #202-Ubuntu SMP Fri Apr 26 12:29:36 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / e0ebb9fdc344956cce029c993a1f60034e0d9904 |
| Default Java | Eclipse Adoptium-17.0.11+9 |
| Max. process+thread count | 82 (vs. ulimit of 30000) |
| modules | C: hbase-backup U: hbase-backup |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6131/1/console |
| versions | git=2.34.1 maven=3.9.8 spotbugs=4.7.3 |
| Powered by | Apache Yetus 0.15.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime |  Logfile | Comment |
|:----:|----------:|--------:|:--------:|:-------:|
| +0 :ok: |  reexec  |   0m 38s |  |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  3s |  |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --author-ignore-list --blanks-eol-ignore-file --blanks-tabs-ignore-file --quick-hadoopcheck  |
|||| _ Prechecks _ |
|||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   3m  6s |  |  master passed  |
| +1 :green_heart: |  compile  |   0m 21s |  |  master passed  |
| +1 :green_heart: |  javadoc  |   0m 16s |  |  master passed  |
| +1 :green_heart: |  shadedjars  |   5m 23s |  |  branch has no errors when building our shaded downstream artifacts.  |
|||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   2m 57s |  |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 21s |  |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 21s |  |  the patch passed  |
| +1 :green_heart: |  javadoc  |   0m 15s |  |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   5m 22s |  |  patch has no errors when building our shaded downstream artifacts.  |
|||| _ Other Tests _ |
| +1 :green_heart: |  unit  |  11m  5s |  |  hbase-backup in the patch passed.  |
|  |   |  30m 48s |  |  |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.46 ServerAPI=1.46 base: https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6131/1/artifact/yetus-jdk17-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/6131 |
| Optional Tests | javac javadoc unit compile shadedjars |
| uname | Linux c074a040592e 5.4.0-177-generic #197-Ubuntu SMP Thu Mar 28 22:45:47 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / e0ebb9fdc344956cce029c993a1f60034e0d9904 |
| Default Java | Eclipse Adoptium-17.0.11+9 |
|  Test Results | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6131/1/testReport/ |
| Max. process+thread count | 3455 (vs. ulimit of 30000) |
| modules | C: hbase-backup U: hbase-backup |
| Console output | https://ci-hbase.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-6131/1/console |
| versions | git=2.34.1 maven=3.9.8 |
| Powered by | Apache Yetus 0.15.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by 2005hithlj** [comment]
This PR for  branch master and branch-3. 
@DieterDP-ng   @Apache9   Could you take a look? Thanks.

File: hbase-backup/src/main/java/org/apache/hadoop/hbase/backup/RestoreDriver.java [hunk 1]
```diff
@@ -124,8 +124,8 @@ private int parseAndRun() throws IOException {
 
     if (cmd.hasOption(OPTION_YARN_QUEUE_NAME)) {
       String queueName = cmd.getOptionValue(OPTION_YARN_QUEUE_NAME);
-      // Set system property value for MR job
-      System.setProperty("mapreduce.job.queuename", queueName);
+      // Set MR job queuename to configuration
+      getConf().set("mapreduce.job.queuename", queueName);
```
Comment (NihalJain) [inline hbase-backup/src/main/java/org/apache/hadoop/hbase/backup/RestoreDriver.java:128]: Hi @2005hithlj  I was curious with one doubt? Does not simply passing -Dmapreduce.job.queuename=YOUR_QUEUE_NAME as args to the MapReduce job work?

File: hbase-backup/src/main/java/org/apache/hadoop/hbase/backup/impl/BackupCommands.java [hunk 2]
```diff
@@ -142,6 +142,12 @@ public void execute() throws IOException {
         throw new IOException(INCORRECT_USAGE);
       }
 
+      if (cmdline.hasOption(OPTION_YARN_QUEUE_NAME)) {
+        String queueName = cmdline.getOptionValue(OPTION_YARN_QUEUE_NAME);
+        // Set MR job queuename to configuration
+        getConf().set("mapreduce.job.queuename", queueName);
```
Comment (NihalJain) [inline hbase-backup/src/main/java/org/apache/hadoop/hbase/backup/impl/BackupCommands.java:148]: Also isn't this change need for BackupDriver? I do see it is being parsed there.

**Comment by NihalJain** [comment]
Also could you please how this PR was tested? I am not sure if writing a UT would be straightforward, if possible that would be great!

**Review by Apache9 — APPROVED** [review]

File: hbase-backup/src/main/java/org/apache/hadoop/hbase/backup/RestoreDriver.java (hunk 1 above)
Comment (2005hithlj) [reply hbase-backup/src/main/java/org/apache/hadoop/hbase/backup/RestoreDriver.java:128]: @NihalJain  Thanks for your review. Currently, the backup command uses the -q option to specify the yarn queue, but it does not take effect, this PR is intended to fix this problem. Additionally, the -D way is not intuitive for users, and we do not have any doc to explain how to use this approach effectively.

File: hbase-backup/src/main/java/org/apache/hadoop/hbase/backup/impl/BackupCommands.java (hunk 2 above)
Comment (2005hithlj) [reply hbase-backup/src/main/java/org/apache/hadoop/hbase/backup/impl/BackupCommands.java:148]: Currently, all the parameter parsing for the backup command is done in BackupCommands.java. If we separately move the parsing of the yarn queue to BackupDriver.java, the code might not be as elegant.

Comment (NihalJain) [reply hbase-backup/src/main/java/org/apache/hadoop/hbase/backup/impl/BackupCommands.java:148]: Got it, thanks for the clarification!

**Review by NihalJain — APPROVED** [review]

**Comment by 2005hithlj** [comment]
@NihalJain Thanks for your review. @DieterDP-ng has verified "Setting the queue through the configuration does work.", and the changes here is also simple and clear. Besides, the UT for this part is not easy to implement. Could you give me some suggestions?

**Pull request merged**